Disc update

taherbert opened the pull request:
Updating a few pieces of logic and cleaning up some of the spell data. Also adds a few missing spells.

taherbert committed:
Update discipline

taherbert committed:
uncomment

taherbert committed:
unused content

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -145,7 +144,7 @@ spec:RegisterPvpTalents( {
 -- Auras
 spec:RegisterAuras( {
-    apathy = {
+        apathy = {
```
Fix indentation.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -160,7 +159,7 @@ spec:RegisterAuras( {
         max_stack = 1
     },
-    body_and_soul = {
+        body_and_soul = {
```
Fix indentation.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -176,6 +175,11 @@ spec:RegisterAuras( {
         max_stack = 1
     },
     death_and_madness = {
+        id = 321973,
```
321973 is not a visible/interactive aura from an API standpoint.  Why is this needed vs. what is included as death_and_madness_debuff already?

taherbert replied:
I think you're right -- removed.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -209,7 +213,7 @@ spec:RegisterAuras( {
         id = 373181,
         duration = 3600,
-        max_stack = 10
+        max_stack = function() return 8/talent.harsh_discipline.rank end,
```
Formatting:
function() return 8 / talent.harsh_discipline.rank end,

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -309,12 +314,12 @@ spec:RegisterAuras( {
         max_stack = 1
     },
-    schism = {
+        schism = {
```
Fix indent.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -309,12 +314,12 @@ spec:RegisterAuras( {
         max_stack = 1
     },
-    shackle_undead = {
+        shackle_undead = {
```
Fix indent.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -454,19 +483,20 @@ spec:RegisterAbilities( {
         talent = "divine_star",
         startsCombat = true,
-        texture = 537026,
+        texture = function() return buff.shadow_covenant.up and 122121 or 110744 end,
```
These aren't texture IDs.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -555,40 +606,96 @@ spec:RegisterAbilities( {
         cooldown = 9,
         recharge = function()
-            if talent.dark_indulgence.enabled then return 9 end
+            if talent.dark_indulgence.enabled then return 9*haste end
```
Formatting:
`if talent.dark_indulgence.enabled then return 9 * haste end`

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -555,40 +606,96 @@ spec:RegisterAbilities( {
             if talent.harsh_discipline.enabled then
-                if buff.harsh_discipline.up and buff.harsh_discipline.stack == 9 then
+                addStack( "harsh_discipline" )
+                if buff.harsh_discipline.up and buff.harsh_discipline.stack == 8/talent.harsh_discipline.rank then
```
`if buff.harsh_discipline.up and buff.harsh_discipline.stack == buff.harsh_discipline.max_stack - 1 then`

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -555,40 +606,96 @@ spec:RegisterAbilities( {
+    },
 
-            if talent.inescapable_torment.enabled and pet.fiend.active then pet.fiend.expires = pet.fiend.expires + 1 end
+    mindbender = {
```
Do not add a Discipline version of Mindbender.  You can add Discipline specific considerations to mindbender in PriestShadow.lua if needed.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -555,40 +606,96 @@ spec:RegisterAbilities( {
         end,
     },
 
+    mindgames = {
```
Does not need to be duplicated if provided in another Priest module.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -631,15 +739,56 @@ spec:RegisterAbilities( {
     },
 
 
+    power_infusion = {
```
Does not need to be added if provided by another Priest specialization module.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -686,8 +835,41 @@ spec:RegisterAbilities( {
             end
         end,
     },
+    -- Shields an ally for 15 sec, absorbing 25,497 damage.
```
Does not need to be added if provided by another Priest specialization module.

taherbert replied:
Removed in favor of Shadow's version, but updated to reflect that weakened soul is no longer applied.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -697,22 +879,30 @@ spec:RegisterAbilities( {
-                if buff.harsh_discipline.up and buff.harsh_discipline.stack == 9 then
-                    removeBuff( "harsh_discipline" )
+                addStack( "harsh_discipline" )
+                if buff.harsh_discipline.stack == 8/talent.harsh_discipline.rank then
```
Like the other case, don't want to addStack to decide to removeBuff.

`if buff.harsh_discipline.stack == buff.harsh_discipline.max_stack - 1 then`

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -845,32 +1036,109 @@ spec:RegisterAbilities( {
     },
 
-    -- Smites an enemy for 1,315 Holy damage.
+    shadowfiend = {
```
Do not add to Discipline; it exists elsewhere.

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -845,32 +1036,109 @@ spec:RegisterAbilities( {
     smite = {
         id = 585,
-        cast = 1.5,
+        cast = function() return 1.5*haste end,
```
This should already be hasted automatically.

Try `/dump Hekili.Class.abilities.smite.cast` in-game to verify before making this change.

taherbert replied:
You're correct!

Hekili commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -845,32 +1036,109 @@ spec:RegisterAbilities( {
             if talent.harsh_discipline.enabled then
-                if buff.harsh_discipline.up and buff.harsh_discipline.stack == 9 then
+                addStack( "harsh_discipline" )
+                if buff.harsh_discipline.stack == 8/talent.harsh_discipline.rank then
```
Same as previous comments re: Harsh Discipline.

Hekili reviewed (commented):
Overall, some minor formatting changes.

If a spell exists in another module and isn't significantly different, just add Discipline's mechanics to the other spell.  It helps avoid issues where spells get out of sync.  Shadowfiend and Mindbender involve some keybind swapping mechanics but otherwise share the same entry.

taherbert commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -533,10 +566,28 @@ spec:RegisterAbilities( {
         copy = 120644
     },
 
+    holy_nova = {
```
@Hekili removing holy nova as well, duplicative with shadow implementation

taherbert commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -845,32 +1036,109 @@ spec:RegisterAbilities( {
+    },
+
+
+    shadow_word_death = {
```
Updating Shadow's implementation instead

taherbert commented on Dragonflight/PriestDiscipline.lua:
```diff
@@ -899,8 +1167,8 @@ spec:RegisterAbilities( {
     vampiric_embrace = {
```
Removed in favor of Shadow's implementation

taherbert committed:
Resolve comments

taherbert commented:
@Hekili thanks for the feedback, addressed your comments! Moved some of the logic to Shadow's implementation instead and removed some abilities that are unlikely to ever be used in any kind of rotation (e.g. Pain Suppression).

taherbert committed:
add copy to resolve error message

taherbert committed:
make sure to add harsh_discipline stack after creating _ready

Hekili commented:
> @Hekili thanks for the feedback, addressed your comments! Moved some of the logic to Shadow's implementation instead and removed some abilities that are unlikely to ever be used in any kind of rotation (e.g. Pain Suppression).

PI can stay; it's conceivable somebody might put it as a self-defensive or something in a custom priority.

taherbert committed:
add back PI

taherbert commented:
> PI can stay

Sure, re-added @Hekili

Hekili committed:
Update PriestDiscipline.lua

Formatting and Harsh Discipline fixes.

Hekili committed:
Update PriestShadow.lua

Make some changes work for both Discipline and Shadow.

Hekili reviewed (commented):
I made a few changes to compatibility across specs.
